modules/navier_stokes/src/materials/INSADMaterial.C

```diff
@@ -66,24 +64,70 @@ INSADMaterial::INSADMaterial(const InputParameters & parameters)
   }
   else
     _gravity_set = false;
-  if (getParam<bool>("include_viscous_term_in_strong_form"))
-    mooseError("Sorry no TypeNTensor operations are currently implemented, so we cannot add the "
-               "strong form contribution of the viscous term. Note that for linear elements, this "
-               "introduces no error, and in general for bi-linear elements, the error is small");
 }
 
 void
 INSADMaterial::computeQpProperties()
 {
   _mass_strong_residual[_qp] = -_grad_velocity[_qp].tr();
+  if (_coord_sys == Moose::COORD_RZ)
+    // Subtract u_r / r
+    _mass_strong_residual[_qp] -= _velocity[_qp](0) / _q_point[_qp](0);
+
   _convective_strong_residual[_qp] = _rho[_qp] * _grad_velocity[_qp] * _velocity[_qp];
   _td_strong_residual[_qp] =
       _transient_term ? _rho[_qp] * (*_velocity_dot)[_qp] : ADRealVectorValue(0);
   _gravity_strong_residual[_qp] = _gravity_set ? -_rho[_qp] * _gravity : ADRealVectorValue(0);
   _mms_function_strong_residual[_qp] = -RealVectorValue(_x_vel_fn.value(_t, _q_point[_qp]),
                                                         _y_vel_fn.value(_t, _q_point[_qp]),
                                                         _z_vel_fn.value(_t, _q_point[_qp]));
+
+  // // The code immediately below is fictional. E.g. there is no Moose::Laplacian nor is there
+  // // currently a _second_velocity member because TypeNTensor (where N = 3 in this case) math is not
+  // // really implemented in libMesh. Hence we cannot add this strong form contribution of the viscous
+  // // term at this time. Note that for linear elements this introduces no error in the consistency of
+  // // stabilization methods, and in general for bi-linear elements, the error introduced is small
+  // _viscous_strong_residual[_qp] = -_mu[_qp] * Moose::Laplacian(_second_velocity[_qp]);
+
+  if (_coord_sys == Moose::COORD_RZ)
+  {
```

Comment:
@lindsayad Could you confirm with me: for RZ, _viscous_strong_residual is only scalar part, but not vector part, for Cartesian, it is basically zero. So it means if we use stabilization, it causes small error for bi-linear elements.

Reply:
There is a piece missing for both Cartesian and cylindrical coordinates and that is the "Cartesian" Laplacian, e.g any second derivative terms.

I may have to update that comment...I seem to recall a long while ago testing the Cartesian AD stabilization method and discovering that for bilinears the error was zero, and that you would only incur error if using biquadratics. I believe this is because the Laplacian includes only the xx, yy, zz derivatives which are zero for bilinears (but wouldn't be zero for biquadratics)

Reply:
It could be no error for bilinear case. I am curious how do you check if there is any error that is coming the stabilization Lap term.